Update logging configuration to use dynamic log level from settings

jmrtnz94 opened the pull request:
### **PR Type**
Enhancement, Configuration changes


___

### **Description**
- Updated logging configuration to use dynamic log levels.

- Added a default log level setting in `configuration.toml`.

- Improved flexibility by fetching log level from settings.


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>gitlab_webhook.py</strong><dd><code>Use dynamic log level in logger setup</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/servers/gitlab_webhook.py

<li>Changed logger setup to use dynamic log level.<br> <li> Default log level set to "INFO" if not configured.


</details>


  </td>
  <td><a href="https://github.com/qodo-ai/pr-agent/pull/1667/files#diff-c3ea1ba460930c6e6e6a2c36a7dc729d79730a96e872d40ffb246f80bc7c0880">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>configuration.toml</strong><dd><code>Add default log level to configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/settings/configuration.toml

<li>Added a <code>log_level</code> setting with default value "INFO".<br> <li> Enhanced configuration to support dynamic logging levels.


</details>


  </td>
  <td><a href="https://github.com/qodo-ai/pr-agent/pull/1667/files#diff-66cfda5143e484ee53ecf7aa0df7dca8ad0b181256f4b0675905db35bcbbae78">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

jmrtnz94 commented:
This should resolved https://github.com/qodo-ai/pr-agent/issues/1592.

jmrtnz94 commented:
Let me know if this implementation works. I can update the other webhook servers as well.

Also let me know if I should update any documentation within the project.

mrT23 commented:
Hi @jmrtnz94 
thanks for the PR.

The default should be 'DEBUG', as it was before (you can change it on your local deployment)

And if you can similarly update the github and bitbucket apps, it will be great

jmrtnz94 commented:
Hey @mrT23,

Good thought on keeping the original log level - I've set that back to "DEBUG". I also added the same changes to the other servers. And I updated the documentation so that devs know about this configuration.

mrT23 commented:
great, thanks for the PR

jmrtnz94 commented:
> great, thanks for the PR

Glad to help out! Even though it was a simple change 😅.
